### PR TITLE
Attempt to support Spring Boot WebApplicationContext initialization and lookup

### PR DESCRIPTION
**Pull request by lincolnthree**
Fixes https://github.com/ocpsoft/rewrite/issues/362

### CODE REVIEW

File: integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java [hunk 1]
```diff
@@ -18,24 +18,34 @@
...
  * {@link ServiceEnricher} implementation for Spring.
  * 
  * @author Christian Kaltepoth
  */
-public class SpringServiceEnricher implements ServiceEnricher
+public class SpringServiceEnricher implements ServiceEnricher, ContextListener
 {
 
    private final Logger log = Logger.getLogger(SpringServiceEnricher.class);
+   private static ServletContext context;
```
Comment (larsgrefer) [inline integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java:38]: What happens, if this class runs in a traditionally deployed war file in a servlet container?

If there were multiple applications deployed, the ServletContext instances could overwrite each other and possibly leak into the wrong application?

Comment (lincolnthree) [reply integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java:38]: You mean in Servlet Container where someone has put shared dependencies in the servlet container classpath? Or in an EAR type container where JARs and dependencies are shared between WARs? I guess that could be a concern. Let me think about that.

File: integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java [hunk 2]
```diff
@@ -18,24 +18,34 @@
...
+import org.ocpsoft.rewrite.servlet.spi.ContextListener;
 import org.springframework.web.context.support.SpringBeanAutowiringSupport;
 
 /**
  * {@link ServiceEnricher} implementation for Spring.
  * 
  * @author Christian Kaltepoth
  */
-public class SpringServiceEnricher implements ServiceEnricher
+public class SpringServiceEnricher implements ServiceEnricher, ContextListener
```
Comment (larsgrefer) [inline integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java:34]: With this setup, the `SpringServiceEnricher` will be instantiated twice, won't it? Once as `ServiceEnricher` and once as `ContextListener`

Comment (lincolnthree) [reply integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java:34]: Yes, but only one of them would be used as an Enricher. The other would just handle the static. I could split this into a different ContextListener implementation, and try to deal with the EAR / shared lib classloader issue there.

**Comment by lincolnthree** [comment]
@larsgrefer Ok. Updated with what I think should resolve the issue. Let me know your thoughts!

**Comment by lincolnthree** [comment]
Dangit. I did not mean to close this by renaming the branch. Working on a fix.

**Pull request closed without merging**